presto-main/src/main/java/com/facebook/presto/operator/unnest/OrdinalityBlockBuilder.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.unnest;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.LongArrayBlock;
+
+import java.util.Optional;
+
+/**
+ * This class manages the details for building ordinality channel blocks without copying input data
+ */
+class OrdinalityBlockBuilder
+{
+    private OrdinalityBlockBuilder()
+    {}
+
+    public static Block buildOutputBlock(int[] maxEntries, int offset, int length, int totalEntriesForBatch)
```

Comment:
I'd inline this method and remove this class.

Reply:
Moved the method to UnnestOperator.